scaletest/dashboard/chromedp.go

```diff
@@ -0,0 +1,181 @@
+	"template_versions":          `a[href^="/templates/"][href$="/versions"]:not([aria-current])`,
+	"template_embed":             `a[href^="/templates/"][href$="/embed"]:not([aria-current])`,
+	"template_insights":          `a[href^="/templates/"][href$="/insights"]:not([aria-current])`,
+}
+
+// ClickRandomElement returns an action that will click an element from the given selectors at random.
+// If no elements are found, an error is returned.
+// If more than one element is found, one is chosen at random.
+// The label of the clicked element is returned.
+func ClickRandomElement(ctx context.Context) (Label, Action, error) {
+	var matched Selector
+	var matchedLabel Label
+	var found bool
+	var err error
+	for l, s := range defaultSelectors {
+		matched, found, err = randMatch(ctx, s)
+		if err != nil {
+			return "", nil, xerrors.Errorf("find matches for %q: %w", s, err)
+		}
+		if !found {
+			continue
+		}
+		matchedLabel = l
+		break
+	}
+	if !found {
+		return "", nil, xerrors.Errorf("no matches found")
+	}
+
+	return "click_" + matchedLabel, func(ctx context.Context) error {
+		if err := clickAndWait(ctx, matched); err != nil {
+			return xerrors.Errorf("click %q: %w", matched, err)
+		}
+		return nil
+	}, nil
+}
+
+// randMatch returns a random match for the given selector.
+// The returned selector is the full XPath of the matched node.
+// If no matches are found, an error is returned.
+// If multiple matches are found, one is chosen at random.
+func randMatch(ctx context.Context, s Selector) (Selector, bool, error) {
+	var nodes []*cdp.Node
+	err := chromedp.Run(ctx, chromedp.Nodes(s, &nodes, chromedp.NodeVisible, chromedp.AtLeast(0)))
+	if err != nil {
+		return "", false, xerrors.Errorf("get nodes for selector %q: %w", s, err)
+	}
+	if len(nodes) == 0 {
+		return "", false, nil
+	}
+	n := pick(nodes)
+	return Selector(n.FullXPath()), true, nil
+}
+
+// clickAndWait clicks the given selector and waits for the page to finish loading.
+// The page is considered loaded when the network event "LoadingFinished" is received.
+func clickAndWait(ctx context.Context, s Selector) error {
+	return chromedp.Run(ctx, chromedp.Tasks{
+		chromedp.Click(s, chromedp.NodeVisible),
+		chromedp.ActionFunc(func(ctx context.Context) error {
+			return waitForEvent(ctx, func(e interface{}) bool {
+				if _, ok := e.(*network.EventLoadingFinished); ok {
+					return true
+				}
+				return false
+			})
+		}),
+	})
+}
+
+// initChromeDPCtx initializes a chromedp context with the given session token cookie
+//
+//nolint:revive // yes, headless is a control flag
+func initChromeDPCtx(ctx context.Context, u *url.URL, sessionToken string, headless bool) (context.Context, context.CancelFunc, error) {
+	dir, err := os.MkdirTemp("", "scaletest-dashboard")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	allocOpts := append(chromedp.DefaultExecAllocatorOptions[:],
+		chromedp.UserDataDir(dir),
+		chromedp.DisableGPU,
```

Comment:
nit: I guess that this will switch Terminal to the fallback mode?

Reply:
Probably, this can be an option though. I've just heard reports that `DisableGPU` makes it more reliable.